content/Registration and Authentication in your Django app with dj-rest-auth part1- Bruno Michetti.md

```diff
@@ -0,0 +1,214 @@
+## Registration and Authentication in your Django app with dj-rest-auth (part 1)
+
+A huge amount of existing applications have registration and authentication for users. Maybe every developer in the world has implemented something related to this in their work or while they learned. After the creation of the [Django REST framework](https://www.django-rest-framework.org/), Django developers started to implement more and more app-level REST API endpoints.  
+As a result, an open source package called [dj-rest-auth](https://dj-rest-auth.readthedocs.io/en/latest/index.html) has been developed to provide a set of REST API endpoints to handle user registration and authentication tasks. In this article divided in two parts, I will talk about this package; and given that it can generate some problems at first, I will give you some tips to avoid them.
+
+### The dj-rest-auth package
+
+If you are a Django developer and you are coding a REST API with authentication, you will find the django-rest-auth package very useful. As said, this package provides a set of REST API endpoints to manage user registration and authentication. After an easy [installation and configuration](https://dj-rest-auth.readthedocs.io/en/latest/installation.html), you will have endpoints for:
```

Comment:
```suggestion
If you are a Django developer and you are coding a REST API with authentication, you will find the django-rest-auth _library_ very useful. As I said, this _library_ provides a set of REST API endpoints to manage user registration and authentication. After an easy [installation and configuration](https://dj-rest-auth.readthedocs.io/en/latest/installation.html), you will have endpoints for:
```
I think this is a library, not a package, check this https://www.geeksforgeeks.org/what-is-the-difference-between-pythons-module-package-and-library/

Reply:
Check the other phrases that contain `package`

Reply:
You're right, it's a library. Thanks for that link!

Reply:
I'm not quite sure of calling it package, please take a look a this:
https://docs.djangoproject.com/en/3.1/intro/reusable-apps/#reusability-matters

```
Package? App?

A Python package provides a way of grouping related Python code for easy reuse. A package contains one or more files of Python code (also known as “modules”).

A package can be imported with import foo.bar or from foo import bar. For a directory (like polls) to form a package, it must contain a special file __init__.py, even if this file is empty.

A Django application is a Python package that is specifically intended for use in a Django project. An application may use common Django conventions, such as having models, tests, urls, and views submodules.

Later on we use the term packaging to describe the process of making a Python package easy for others to install. It can be a little confusing, we know.
```

It says that a django app (just as dj-rest-auth) it's  aPython package that is specifically intended for use in a Django project. WDYT?